raiden/storage/utils.py

```diff
@@ -1,3 +1,44 @@
+"""
+Notes:
+
+Every primary key *MUST* also have *NOT NULL* specified [1].
+
+`WITHOUT ROWID` [2] was considered, in order to force sqlite3 to use "proper"
+clustered indexes and to enforce the NOT NULL constraint for primary keys.
+However the write/read pattern for Raiden does not fall under the recommended
+categories, and some important features of the database won't work with that
+flag. Instead of `WITHOUT ROWID` the implicit rowid column is used.
+
+Using an explicit `INTEGER PRIMARY KEY` [3] was considered, however 64 bits
+are too few to fit a well ordered ID with nanosecond precision and a random
+tag.
+
+The primary key is *not* auto-increment because the IDs are populated with
+ULIDs [7]. ULIDs are necessary because the application must know the ID of
+elements inserted in bulk, and this cannot be know reliably if auto increment
+is used, even with synchronizaiton over `lastrowid`. Additionally random IDs
+are less likely to collide with is useful to detect programming errors. Typos
+are a good example that can have catastrophic results, where REPLACE/DELETE
+queries can operate on the wrong table or column and with the possibility of
+collision change the wrong data.
+
+ULIDs are necessary over UUIDs because they are lexicographically sortable by
+design, and state changes must be saved in order. This has the additional
+benefit of being a good write pattren, even though the implicit clustered index
+based on rowid will prevent large rewrites of the DB's tables, using ULIDs
+prevent rewrites of the PK Index.
+
+Manifest typing [4] is used for the ULIDs instead PARSE_COLNAMES [5] because
+of the easy of convertion.
```

Comment:
```suggestion
of the ease of conversion.
```

BTW: I really like this well written documentation!